archive/d/dart/prime-number.dart

```diff
@@ -0,0 +1,34 @@
+// Issue 4970
+import 'dart:math';
+void main(List<String> args){
+  const String error_message= "Usage: please input a non-negative integer";
+  if (args.isEmpty){
+    print(error_message);
+    return;
+  }
+  try{
+    int num_needed = int.parse(args[0]);
+    if (num_needed.isNegative){
+      print(error_message);
+      return;
+    }
+
+    // If the number is even number other than 2 OR if the number is multiple of 5 other than 5 or if number is either 0 or 1, Print Composite
+    if( (num_needed == 0)|| (num_needed == 1) || (num_needed %2 ==0 && num_needed != 2) || (num_needed %5  == 0 && num_needed != 5) ){
```

Comment:
Again, the check for divisibility by 5 is not necessary. The `for` loop will not run for the value of 5, and any multiple of 5 will be found by the `for` loop.